### PR TITLE
FISH-6523 postbootcommand Python Tests

### PR DESCRIPTION
**Pull request by aubi**
## Description
Adding Python tests for postbootcommand file, starting and stopping Payara server for each test.
The first test builds a reproducer, configures and deploys it via postbootcommand file and then tests, if both configuration ways (environment variable and MP config) work.
The second test creates H2 JDBC pool and checks, if it is available on the server.

### Blockers
We need to figure out a good directory name for this type of tests.

### New tests
appserver/tests/complex/postboot/testPostbootcommand.py

### Testing Performed
This test runs 2 separated tests, one with a Java reproducer, one creating only JDBC pool

### Testing Environment
Linux, OpenJDK 11

## Notes for Reviewers
Build Payara, run testPostbootcommand.py. It should start and shutdown Payara two times and end up with message "0 errors."

### CODE REVIEW

File: appserver/tests/complex/postboot/testPostbootcommand.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/python3
+import os
+import subprocess
+import time
+import urllib.request
+import testpayaralib
+
+# Setup
+testpayaralib.PAYARA_PATH_BIN = os.path.abspath("../../../../appserver/distributions/payara/target/stage/payara6/bin")
```
Comment (Pandrex247) [inline appserver/tests/complex/postboot/testPostbootcommand.py:9]: What if I wanted to test this against web profile or a server not located in the default build dir (e.g. like in the Jenkins pipeline).
Could we add some means of reading in user input?

**Review by Pandrex247 — COMMENTED** [review]
Windows doesn't seem to like your test.

```
Current path: C:\Users\AndrewPielage\Git\Payara2\appserver\tests\complex\postboot
Build ConfigurableCDI
Traceback (most recent call last):
  File "C:\Users\AndrewPielage\Git\Payara2\appserver\tests\complex\postboot\testPostbootcommand.py", line 17, in <module>
    errors += testpayaralib.doOneTest("ConfigurableCDI", "ConfigurableCDI", "ConfigurableCDI-1.0-SNAPSHOT", "war", [
  File "C:\Users\AndrewPielage\Git\Payara2\appserver\tests\complex\postboot\testpayaralib.py", line 25, in doOneTest
    subprocess.run(['mvn', 'clean', 'package'])
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2288.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2288.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2288.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 1440, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

**Comment by Pandrex247** [comment]
I vote for naming the directory _functional_, _integration_, or _smoke_ rather than _complex_.
My reasoning being how do you define a complex test?

Tests that are to end up in here are more akin to just "not arquillian" / "not maven" tests.

**Review by simonladen — COMMENTED** [review]
I only have comments regarding the names: the function in testpayaralib.py. 
Also, the file testPostbootcommand.py may need to be testpostbootcommand.py to follow Python naming conventions.

File: appserver/tests/complex/postboot/testPostbootcommand.py (hunk 1 above)
Comment (aubi) [reply appserver/tests/complex/postboot/testPostbootcommand.py:9]: I kept it simple for beginning, but you are right. Added.

**Comment by aubi** [comment]
Jenkins test please

**Comment by aubi** [comment]
Jenkins test please

**Review by simonladen — APPROVED** [review]

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
Still doesn't work on Windows - README should at least be updated to say that it doesn't

**Comment by aubi** [comment]
@Pandrex247 This version worked for me (TM), try it again, please.
I had to simplify the second test as juggling with backslashes led only to dead-ends.

**Review by Pandrex247 — COMMENTED** [review]
Getting inconsistent results - validated on multiple Windows machines:

Run 1: hang at http://localhost:4848/management/domain/resources/list-jdbc-connection-pools
Run2: pass
Run 3: fail at http://localhost:4848/management/domain/resources/list-jdbc-connection-pools
Run 4: pass
Run 5: hang at http://localhost:4848/management/domain/resources/list-jdbc-connection-pools

The test also doesn't clean up after itself, making repeat runs sketchy - _testpool_ persists in the domain.xml after the test, meaning the command to create it in the post boot will always fail on subsequent runs without cleaning the domain.xml (which interestingly doesn't break the test)

**Review by Pandrex247 — APPROVED** [review]
🥳

**Comment by aubi** [comment]
Jenkins test please

**Pull request merged**